### DISCUSSION TITLE
setRecoilState requires re-rendering

### DISCUSSION THREAD

**Discussion by Leykwan132** [facebookexperimental/Recoil#1982] [Q&A]
const profileData = useRecoilValue(profileState);
  const setProfileUrl = useSetRecoilState(profileState);
  useEffect(() => {
    setProfileUrl(JSON.parse(window.localStorage.getItem("profileData")));
    console.log(profileData)
  }, []);
I am trying to set the atom to the localStorage value when the user refreshes the page, but it seems like setRecoilState requires a re-render which would potentially crash the application. Any solution to this, thanks!

**Comment by drarmstr** [comment]
Setting Recoil state may cause subscribing components to re-render, but there are optimizations which avoid it in some cases such as when changing the state to the same value.  In general, don't rely on the number of renders for semantic reasons.  It's not clear why re-rendering would crash your application without an example.

For syncing atoms with localStorage, consider [atom effects](https://recoiljs.org/docs/guides/atom-effects#local-storage-persistence)
